src/test/java/seedu/rc4hdb/testutil/ResidentDescriptorBuilder.java

```diff
@@ -0,0 +1,117 @@
+package seedu.rc4hdb.testutil;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import seedu.rc4hdb.model.resident.Resident;
+import seedu.rc4hdb.model.resident.ResidentDescriptor;
+import seedu.rc4hdb.model.resident.fields.Email;
+import seedu.rc4hdb.model.resident.fields.Gender;
+import seedu.rc4hdb.model.resident.fields.House;
+import seedu.rc4hdb.model.resident.fields.MatricNumber;
+import seedu.rc4hdb.model.resident.fields.Name;
+import seedu.rc4hdb.model.resident.fields.Phone;
+import seedu.rc4hdb.model.resident.fields.Room;
+import seedu.rc4hdb.model.tag.Tag;
+
+/**
+ * A utility class to help with building ResidentDescriptor objects.
+ */
+public class ResidentDescriptorBuilder {
+
+    private ResidentDescriptor descriptor;
+
+    public ResidentDescriptorBuilder() {
+        descriptor = new ResidentDescriptor();
+    }
+
+    public ResidentDescriptorBuilder(ResidentDescriptor descriptor) {
+        this.descriptor = new ResidentDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code ResidentDescriptor} with fields containing {@code resident}'s details
+     */
+    public ResidentDescriptorBuilder(Resident resident) {
```

Comment:
I'm just wondering the point of this Builder Class and constructor, since this was not used elsewhere in the code. Not necessary to fix!